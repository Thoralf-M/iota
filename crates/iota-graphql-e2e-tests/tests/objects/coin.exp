processed 4 tasks

init:
A: object(0,0)

task 1, lines 7-34:
//# publish --sender A
created: object(1,0), object(1,1), object(1,2), object(1,3), object(1,4), object(1,5)
mutated: object(0,0)
unchanged_shared: 0x0000000000000000000000000000000000000000000000000000000000000403
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 15618000,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, line 36:
//# create-checkpoint
Checkpoint created: 1

task 3, lines 38-86:
//# run-graphql
Response: {
  "data": {
    "iotaCoins": {
      "edges": [
        {
          "cursor": "ICmkZlwOHb7A7n/F8ogaDQHNnGeTtDMEgXve7z+gTfsMAQAAAAAAAAA=",
          "node": {
            "coinBalance": "299999983382000",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
              }
            }
          }
        },
        {
          "cursor": "II8mSWvMPFBpfNYm4Jp55EGcry0qahEB13OfM/2O9QOGAQAAAAAAAAA=",
          "node": {
            "coinBalance": "30000000000000000",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
              }
            }
          }
        },
        {
          "cursor": "IOkgV6r+w/sR9+8WHXDAN0keSRfyOoRSeeaEaBEtn7AKAQAAAAAAAAA=",
          "node": {
            "coinBalance": "300000000000000",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
              }
            }
          }
        }
      ]
    },
    "fakeCoins": {
      "edges": [
        {
          "cursor": "IFNo5MdFBXb0/DC2Hu+J1/cH3nJeLlG16cohLpxwz137AQAAAAAAAAA=",
          "node": {
            "coinBalance": "2",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0xae951728f9dd6b1cc060af6c26374bb2b45b4773f088c54f74cd5db8bfa8d580::fake::FAKE>"
              }
            }
          }
        },
        {
          "cursor": "IGLgadFZR57PhSa3qBfnmx38Gm+POLzGlAMNbhG7QtH1AQAAAAAAAAA=",
          "node": {
            "coinBalance": "3",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0xae951728f9dd6b1cc060af6c26374bb2b45b4773f088c54f74cd5db8bfa8d580::fake::FAKE>"
              }
            }
          }
        },
        {
          "cursor": "IHRDETOlvoEKmIPf5ezdO21+fbsrKzCxWq4vVUQ3C1blAQAAAAAAAAA=",
          "node": {
            "coinBalance": "1",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0xae951728f9dd6b1cc060af6c26374bb2b45b4773f088c54f74cd5db8bfa8d580::fake::FAKE>"
              }
            }
          }
        }
      ]
    },
    "address": {
      "coins": {
        "edges": [
          {
            "cursor": "ICmkZlwOHb7A7n/F8ogaDQHNnGeTtDMEgXve7z+gTfsMAQAAAAAAAAA=",
            "node": {
              "coinBalance": "299999983382000",
              "contents": {
                "type": {
                  "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
                }
              }
            }
          }
        ]
      },
      "allBalances": {
        "edges": [
          {
            "cursor": "eyJ0IjoiMHgwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAyOjppb3RhOjpJT1RBIiwiYyI6MX0",
            "node": {
              "coinType": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA"
              },
              "coinObjectCount": 1,
              "totalBalance": "299999983382000"
            }
          },
          {
            "cursor": "eyJ0IjoiMHhhZTk1MTcyOGY5ZGQ2YjFjYzA2MGFmNmMyNjM3NGJiMmI0NWI0NzczZjA4OGM1NGY3NGNkNWRiOGJmYThkNTgwOjpmYWtlOjpGQUtFIiwiYyI6MX0",
            "node": {
              "coinType": {
                "repr": "0xae951728f9dd6b1cc060af6c26374bb2b45b4773f088c54f74cd5db8bfa8d580::fake::FAKE"
              },
              "coinObjectCount": 3,
              "totalBalance": "6"
            }
          }
        ]
      },
      "firstBalance": {
        "edges": [
          {
            "cursor": "eyJ0IjoiMHgwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAyOjppb3RhOjpJT1RBIiwiYyI6MX0"
          }
        ]
      },
      "lastBalance": {
        "edges": [
          {
            "cursor": "eyJ0IjoiMHhhZTk1MTcyOGY5ZGQ2YjFjYzA2MGFmNmMyNjM3NGJiMmI0NWI0NzczZjA4OGM1NGY3NGNkNWRiOGJmYThkNTgwOjpmYWtlOjpGQUtFIiwiYyI6MX0"
          }
        ]
      }
    }
  }
}
