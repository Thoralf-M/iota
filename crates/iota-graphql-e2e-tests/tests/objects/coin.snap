---
source: external-crates/move/crates/move-transactional-test-runner/src/framework.rs
---
processed 4 tasks

init:
A: object(0,0)

task 1, lines 7-34:
//# publish --sender A
created: object(1,0), object(1,1), object(1,2), object(1,3), object(1,4), object(1,5)
mutated: object(0,0)
unchanged_shared: 0x0000000000000000000000000000000000000000000000000000000000000403
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 15618000,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, line 36:
//# create-checkpoint
Checkpoint created: 1

task 3, lines 38-86:
//# run-graphql
Response: {
  "data": {
    "iotaCoins": {
      "edges": [
        {
          "cursor": "IHQYRhDrG+U4Mtk7XANbay/D7RiOuc6/tqZtP2CrXkzWAQAAAAAAAAA=",
          "node": {
            "coinBalance": "300000000000000",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
              }
            }
          }
        },
        {
          "cursor": "INNUgLY/0naiwb5be28kr+hOzRhK6O/eRpeRYJaPdg7rAQAAAAAAAAA=",
          "node": {
            "coinBalance": "299999983382000",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
              }
            }
          }
        },
        {
          "cursor": "IN1dw8JSMZ3fTkObz72dmGJuhvfZkEEkgWwbSuqY+X4/AQAAAAAAAAA=",
          "node": {
            "coinBalance": "30000000000000000",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
              }
            }
          }
        }
      ]
    },
    "fakeCoins": {
      "edges": [
        {
          "cursor": "IGTQkzLYvf6IeIFSqxnStvUjx+SpGkWIVh0SJasNdEFgAQAAAAAAAAA=",
          "node": {
            "coinBalance": "1",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0xb9998bf9b29e67bea12747ec03a8651f1e14ca673e29a12c3b2fd23f19b4da0e::fake::FAKE>"
              }
            }
          }
        },
        {
          "cursor": "IJCciOCIiKOVgRANnkv2Ek+63tGQRaRvBTdoC5n6uBeKAQAAAAAAAAA=",
          "node": {
            "coinBalance": "3",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0xb9998bf9b29e67bea12747ec03a8651f1e14ca673e29a12c3b2fd23f19b4da0e::fake::FAKE>"
              }
            }
          }
        },
        {
          "cursor": "IMgdXp2TGN2R8SESmdKnbHB97gHQlsVG1EhT1QBdKAjgAQAAAAAAAAA=",
          "node": {
            "coinBalance": "2",
            "contents": {
              "type": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0xb9998bf9b29e67bea12747ec03a8651f1e14ca673e29a12c3b2fd23f19b4da0e::fake::FAKE>"
              }
            }
          }
        }
      ]
    },
    "address": {
      "coins": {
        "edges": [
          {
            "cursor": "INNUgLY/0naiwb5be28kr+hOzRhK6O/eRpeRYJaPdg7rAQAAAAAAAAA=",
            "node": {
              "coinBalance": "299999983382000",
              "contents": {
                "type": {
                  "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
                }
              }
            }
          }
        ]
      },
      "allBalances": {
        "edges": [
          {
            "cursor": "eyJ0IjoiMHgwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAyOjppb3RhOjpJT1RBIiwiYyI6MX0",
            "node": {
              "coinType": {
                "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA"
              },
              "coinObjectCount": 1,
              "totalBalance": "299999983382000"
            }
          },
          {
            "cursor": "eyJ0IjoiMHhiOTk5OGJmOWIyOWU2N2JlYTEyNzQ3ZWMwM2E4NjUxZjFlMTRjYTY3M2UyOWExMmMzYjJmZDIzZjE5YjRkYTBlOjpmYWtlOjpGQUtFIiwiYyI6MX0",
            "node": {
              "coinType": {
                "repr": "0xb9998bf9b29e67bea12747ec03a8651f1e14ca673e29a12c3b2fd23f19b4da0e::fake::FAKE"
              },
              "coinObjectCount": 3,
              "totalBalance": "6"
            }
          }
        ]
      },
      "firstBalance": {
        "edges": [
          {
            "cursor": "eyJ0IjoiMHgwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAyOjppb3RhOjpJT1RBIiwiYyI6MX0"
          }
        ]
      },
      "lastBalance": {
        "edges": [
          {
            "cursor": "eyJ0IjoiMHhiOTk5OGJmOWIyOWU2N2JlYTEyNzQ3ZWMwM2E4NjUxZjFlMTRjYTY3M2UyOWExMmMzYjJmZDIzZjE5YjRkYTBlOjpmYWtlOjpGQUtFIiwiYyI6MX0"
          }
        ]
      }
    }
  }
}
